tests/Microsoft.Fx.Portability.MetadataReader.Tests/ManagedMetadataReaderTests.cs

```diff
@@ -245,5 +249,25 @@ public bool IsFrameworkAssembly(AssemblyReferenceInformation assembly)
                 return string.Equals(_assemblyName, assembly?.Name);
             }
         }
+
+        private class StreamFromFileAssemblyFile : IAssemblyFile
+        {
+            private readonly byte[] _bytes;
+            private readonly string _path;
+
+            public StreamFromFileAssemblyFile(string path)
+            {
+                _path = path;
+                _bytes = File.ReadAllBytes(path);
+            }
+
+            public bool Exists { get; } = true;
+
+            public string FullName => _path;
```

Comment:
Yes. I've updated the tests to create various implementations of IAssemblyFile: 

- ResourceStreamAssemblyFile will take a resource name and stream the file from there
- CSharpCompileAssemblyFile will compile an assembly and emit the result to a byte[] that is wrapped in a stream

Neither of these set the Name (there is a test that does set the Name, though). The only places Name is used is to list assemblies it fails to open or non existent files.